src/core/infiniteQueryBehavior.ts

```diff
@@ -153,9 +148,6 @@ export function infiniteQueryBehavior<
         finalPromiseAsAny.cancel = () => {
           cancelled = true
           abortController?.abort()
-          if (isCancelable(promise)) {
-            promise.cancel()
-          }
         }
```

Comment:
I think you _might_ be able to get rid of this internal `.cancel` function by replacing this with

```ts
context.signal?.addEventListener('abort', () => {
  cancelled = true;
  abortController?.abort()
})
```

Reply:
Thanks, this seems to work now, although the signal wasn't being passed down to the `QueryBehavior`'s `FetchContext`, so I added the signal to it similar to how we added it to `QueryFunctionContext`